### DISCUSSION TITLE
Discipline Namespace cross-referencing of classes/attributes

### DISCUSSION THREAD

**Discussion by jordanpadams** [pds-data-dictionaries/PDS-Data-Dictionaries.github.io#15] [Ideas]
@acraugh and I chatted about this offline, and came up with a near-term plan and future ideal design cross-referencing classes/attributes in the PDS4 IM:

**_Near-term:_** Cross-referencing will be allowed across discipline namespaces will require approval by PDS Namespace Change Control Board (or whatever we are calling it), and will be reviewed on a case-by-case basis. The discipline steward requesting the update will have to provide sufficient rationale to the CCB describing the cross-discipline applicability and usage. All classes/attributes enabled for cross-referencing will be clearly documented at TBD top-level cross-referencing documentation that will clearly associate what classes/attributes are exposed and the Discipline Namespaces using them. The EN member of the CCB will be responsible for ensuring any future proposed changes to those classes/attributes trigger a more detailed review process that must include the stewards of the discipline namespaces that may be impacted.
 
**_Future/Ideal:_** Any classes/attributes identified as being cross-discipline should be refactored out of current namespace, and added to its own cross-discipline namespace with improvements to the definitions of those classes/attributes to better enable cross-referencing and reusability.

**Comment by mdrum** [comment]
+1
I don't like the idea of a generic public bucket of reusable classes though, as was previously floated. Each reusable class from a namespace should get its own namespace, even if it's just one or two attributes. The only hard part will be naming it... and deciding what to call this new genre of namespaces.
